pkg/k8s/volumes/crud.go

```diff
@@ -98,8 +108,12 @@ func checkPVCValues(pvc *apiv1.PersistentVolumeClaim, dev *model.Dev, devPath st
 	if !ok {
 		return fmt.Errorf("current okteto volume size is wrong. Run '%s' and try again", utils.GetDownCommand(devPath))
 	}
-	if currentSize.Cmp(resource.MustParse(dev.PersistentVolumeSize())) > 0 {
-		if currentSize.Cmp(resource.MustParse("10Gi")) != 0 || dev.PersistentVolumeSize() != model.OktetoDefaultPVSize {
+	devPVSize, err := resource.ParseQuantity(dev.PersistentVolumeSize())
+	if err != nil {
+		return fmt.Errorf("error parsing dev volume size %q: %w", dev.PersistentVolumeSize(), err)
+	}
+	if currentSize.Cmp(devPVSize) > 0 {
+		if currentSize.Cmp(resource.MustParse("10Gi")) != 0 || dev.HasDefaultPersistentVolumeSize() {
```

Comment:
Wouldn't this return an error if someone deployed a dev volume with previous version (2Gi) and then, it executes it again with the new version (5Gi)? It would return that it doesn't have the default persistent volume size

Reply:
I think it should enter in the other if block since the `currenSize` would be 2Gi and the `devPVSize` would be 5Gi. I don't know what the restart annotation is for though.

One thing I found while investigating this is that we are passing the wrong `pvc` to this function. In `CreateForDev` we should be passing `k8Volume`. As it is now, we are passing the pvc inferred from `dev`, so we are checking against the same value and this function is not doing anything :o

Should I create another ticket to work on that issue or fix it here? If it's just an issue with passing the other parameter, it would be a quick PR and this one could wait for that

Reply:
I've created https://github.com/okteto/okteto/pull/3117 to solve this problem

Reply:
yeah, better to do it in another PR. 

> I think it should enter in the other if block since the currenSize would be 2Gi and the devPVSize would be 5Gi. I don't know what the restart annotation is for though.

You are right, it will enter in the second if. I don't know what the restart annotation does neither. I assume is something needed when the dev volume size increase